src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -204,6 +204,13 @@ void BackgroundNotificationCb (NSNotification obj)
 		void RemoveInflightData (NSUrlSessionTask task, bool cancel = true)
 		{
 			lock (inflightRequestsLock) {
+				InflightData data;
+				if (inflightRequests.TryGetValue (task, out data)) {
+					data = inflightRequests [task];
+					if (cancel)
+						data.CancellationTokenSource.Cancel ();
+					data.Dispose ();
+				}
 				inflightRequests.Remove (task);
```

Comment:
You can move this line inside the `if (inflightRequests.TryGetValue (task, out data)) {` block.

Reply:
Good catch, much better.